src/gpu/ProjectorByBinNiftyPETHelper.cxx

```diff
@@ -0,0 +1,389 @@
+
+    // Copy data from STIR to NiftyPET image
+    unsigned np_z, np_y, np_x, np_1d;
+    for (int z = min_indices[1]; z <= max_indices[1]; z++) {
+        for (int y = min_indices[2]; y <= max_indices[2]; y++) {
+            for (int x = min_indices[3]; x <= max_indices[3]; x++) {
+                // Convert the stir 3d index to a NiftyPET 1d index
+                np_z = unsigned(z - min_indices[1]);
+                np_y = unsigned(y - min_indices[2]);
+                np_x = unsigned(x - min_indices[3]);
+                np_1d = convert_niftypet_im_3d_to_1d_idx(np_x,np_y,np_z);
+                np_vec[np_1d] = stir[z][y][x];
+            }
+        }
+    }
+}
+
+void
+ProjectorByBinNiftyPETHelper::
+convert_image_niftyPET_to_stir(DiscretisedDensity<3,float> &stir, const std::vector<float> &np_vec)
+{
+    // Get the dimensions of the input image
+    Coordinate3D<int> min_indices;
+    Coordinate3D<int> max_indices;
+    int stir_dim[3];
+    get_stir_indices_and_dims(stir_dim,min_indices,max_indices,stir);
+
+    // NiftyPET requires the image to be (z,x,y)=(SZ_IMZ,SZ_IMX,SZ_IMY)
+    // which at the time of writing was (127,320,320).
+    const int np_dim[3] = {SZ_IMZ,SZ_IMX,SZ_IMY};
+    check_im_sizes(stir_dim,np_dim);
+    check_voxel_spacing(stir);
+
+    // Copy data from NiftyPET to STIR image
+    unsigned np_z, np_y, np_x, np_1d;
+    for (int z = min_indices[1]; z <= max_indices[1]; z++) {
+        for (int y = min_indices[2]; y <= max_indices[2]; y++) {
+            for (int x = min_indices[3]; x <= max_indices[3]; x++) {
+                // Convert the stir 3d index to a NiftyPET 1d index
+                np_z = unsigned(z - min_indices[1]);
+                np_y = unsigned(y - min_indices[2]);
+                np_x = unsigned(x - min_indices[3]);
+                np_1d = convert_niftypet_im_3d_to_1d_idx(np_x,np_y,np_z);
+                stir[z][y][x] = np_vec[np_1d];
+            }
+        }
+    }
+}
+
+void
+get_vals_for_proj_data_conversion(std::vector<int> &sizes, std::vector<int> &segment_sequence,
+                                  int &num_sinograms, int &min_view, int &max_view,
+                                  int &min_tang_pos, int &max_tang_pos,
+                                  const ProjData& proj_data, const std::vector<float> &np_vec)
+{
+    shared_ptr<const ProjDataInfoCylindricalNoArcCorr> info_sptr =
+            dynamic_pointer_cast<const ProjDataInfoCylindricalNoArcCorr>(proj_data.get_proj_data_info_sptr());
+    if (is_null_ptr(info_sptr))
+        error("ProjectorByBinNiftyPETHelper: only works with cylindrical projection data without arc-correction");
+
+    const int num_rings     = info_sptr->get_scanner_sptr()->get_num_rings();
+    const int max_ring_diff = info_sptr->get_max_ring_difference(info_sptr->get_max_segment_num());
+
+    segment_sequence.resize(unsigned(2*max_ring_diff+1));
+    sizes.resize(unsigned(2*max_ring_diff+1));
+    segment_sequence[0]=0;
+    sizes[0]=num_rings;
```

Comment:
this code is incorrect. It assumes that the data is span 1 (which it is for the mMR listmode data in the [listmode code](https://github.com/UCL/STIR/blob/e2f0bcb74af6bbf067785e07f58a8430faf3c64d/src/listmode_buildblock/CListRecordECAT8_32bit.cxx#L53), sorry. Instead we need a combination of for instance https://github.com/UCL/STIR/blob/e2f0bcb74af6bbf067785e07f58a8430faf3c64d/src/IO/stir_ecat7.cxx#L1288-L1294 and https://github.com/UCL/STIR/blob/e2f0bcb74af6bbf067785e07f58a8430faf3c64d/src/IO/stir_ecat7.cxx#L1357-L1361:
```C++
segment_sequence[unsigned(0)]=0;
sizes[0]=info_sptr->get_num_axial_poss(0);
for (int segment_num=1; segment_num<=max_segment_num; ++segment_num)
{
   segment_sequence[unsigned(2*segment_num-1)] = -segment_num;
   segment_sequence[unsigned(2*segment_num)] = segment_num;
   sizes [unsigned(2*segment_num-1)] =info_sptr->get_num_axial_poss(-segment_num);
   sizes [unsigned(2*segment_num)] =info_sptr->get_num_axial_poss(segment_num);
```
in fact, we don't really need `sizes` at all, as it's just `info_sptr->get_num_axial_poss(segment_sequence[i])`
```